### ISSUE TITLE
False negative for rules excluded in overrides on unmatched files

### ISSUE THREAD

**Issue by paibamboo** [typescript-eslint/typescript-eslint#1031]
Here are some parts of my `.eslintrc.js`:

```javascript
module.exports = {
  // ...
  "rules": {
     "no-bitwise": "error",
     "@typescript-eslint/tslint/config": [
      "error",
      {
        "rulesDirectory": [...],
        "rules": {
           "jsx-no-lambda": true,
        }
      }
    ]
  },
  "overrides": [
    {
      "files": ["*.stories.tsx"],
      "rules": {
        "no-bitwise": "off",
        "@typescript-eslint/tslint/config": [
          "error",
          {
            "rulesDirectory": [...],
            "rules": {
              "jsx-no-lambda": false
            }
          }
        ]
      }
    }
  ]
}
```

With this config, one would expect that `no-bitwise` and `jsx-no-lambda` rules would be applied on all files except story files. For `no-bitwise`, it works as expected. However when `jsx-no-lambda` rule is violated on non-story files, linting does not complain. So this seems to me that eslint's `overrides` config works but something is weird with `@typescript-eslint/tslint/config`.

Here I have `Button.tsx`

```jsx
  // render function
  <button onClick={() => console.info("test")}> /* linting should complain but it does not */
    {children}
  </button>

  // somewhere else
  console.info(1 | 0); // ESLint: Unexpected use of '|'.(no-bitwise)
```

And `Button.stories.tsx`
```jsx
  <Button onClick={() => console.info("test")} /* linting does not complain which is expected */

  console.info(1 | 0); // linting does not complain which is expected
```

**What did you expect to happen?**
Linting should complain on files that do not match `overrides.files` glob pattern.

**What actually happened?**
It does not.

**Versions**

| package                                   | version |
| ----------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin-tslint` | `2.3.2` |
| `@typescript-eslint/parser`               | `2.3.2` |
| `TypeScript`                              | `3.6.3` |
| `ESLint`                                  | `6.5.1` |
| `node`                                    | `10.10.0` |
| `npm`                                     | `6.4.1` |

**Reproducing steps**
- Clone this [repo](https://github.com/crazyfactory/ts-react-boilerplate/)
- `git checkout reproducing-eslint-typescript-parser-bug`
- `npm install`
- `npm run lint`
- Please see `src/app/components/Button.tsx`. This file violates both `no-bitwise` and `jsx-no-lambda`, but only `no-bitwise` is complained about.

**Comment by paibamboo** [comment]
With latest linting packages, this is fixed.
